scripts/events_module/relationship/pregnancy_events.py

```diff
@@ -966,6 +975,99 @@ def get_kits(
             blood_parent.outside = True
             clan.unknown_cats.append(blood_parent.ID)
 
+        # check for more extended family members to create relationships with
+        for kit in all_kitten:
+            for cat_id in kit.get_relatives():
+                relative_cat = Cat.fetch_cat(cat_id)
+                if relative_cat.dead or relative_cat.outside:
+                    continue
+                parents = kit.get_parents()
+                if cat_id in parents:
+                    continue
+                rel_reflection = game.config["new_cat"]["ext_relative_modifier"]
+                y = random.randrange(-5, 10)
+
+                # this finds what the relative's relationship is toward each parent and applies a reflection of that
+                # relationship to the kit. reflection values will be divided by 4 by default and then modified
+                # by the random y value
+                for parent_id in parents:
+                    try:
+                        relation_toward_parent = relative_cat.relationships[parent_id]
+                    except KeyError:
+                        continue
+                    relation_toward_kit = relative_cat.relationships[kit.ID]
```

Comment:
Thinking out loud here, it might be beneficial to look at whether we can make the relationship class iterable over the attributes.